src/zenml/stack_stores/models/stack_wrapper.py

```diff
@@ -0,0 +1,36 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+from typing import List
+
+from pydantic import BaseModel
+
+from zenml.stack import Stack
+from zenml.stack_stores.models import StackComponentConfiguration
+
+
+class StackWrapper(BaseModel):
```

Comment:
You're right that they both are serializable stack representations, but the difference is a bit larger that you make it out. The old `StackConfiguration` which is here mainly for compatibility with the old-style repository `config.yaml` file deserialization only contains the names of all stack components (as you'd find it in that file), whereas the Wrapper basically has the full configuration of the components included as well, so flavor, uuid, config parameters. 

I'd be happy to get rid of the former, especially since it seems to me that the custom `contains_component()` method is never even used anywhere in our project, but that does mean we lose a little bit of validation of stacks stored in yaml (my idea would be to replace that class with a simple `Dict[str, str]`. What do you think?

Reply:
Ah you're right, the StackWrapper contains much more information. In that case the Dict or keeping the super simple old class doesn't make too much of a difference in my opinion

Reply:
Having done the refactor, I think the dict is actually better. Why? Because we don't need to maintain new possible stack components in two places, just adding it to the enum is enough to make the dict work with the new type, and also because we then don't need to check the optional stack components for None-ness, they simply won't appear in the dict if not set.